core/shared/src/main/scala/interface.scala

```diff
@@ -452,3 +452,11 @@ object MagnoliaUtil {
     values.toList.collect { case Left(v) => v }
 
 }
+
+/**
+ * Marker trait for annotations which when encountered on either a parameter or a constructor,
+ * the derivation will not look for a typeclass instance, instead bind the typeclass parameter
+ * on the interface to the 'bind[A]: Typeclass[A]' method of the enclosing class.
+ *
+ * */
+trait RuntimeBoundTypeclass
```

Comment:
Should this be self-typed to `StaticAnnotation`?, i.e.
```scala
trait RuntimeBoundTypeclass { this: StaticAnnotation =>
```
(I'm basing that only on what you said in the PR description!)

Reply:
I might also suggest a better name for this. If we use the `fallback` method instead of `bind`, could we call it `FallbackDerivation` or just `Fallback`?